### ISSUE TITLE
windbg.flare - The remote file either doesn't exist, is unauthorized, or is forbidden for url 

### ISSUE THREAD

**Issue by darioongit** [fireeye/flare-vm#344]
The remote file either doesn't exist, is unauthorized, or is forbidden for url 'https://codemachine.com/downloads/win10rs1/X86%20Debuggers%20And%20Tools-x86_en-us.msi'. Exception calling "GetResponse" with "0" argument(s): "The remote server returned an error: (404) Not Found."
Downloading windbg.flare
  from 'https://codemachine.com/downloads/win10rs1/X86%20Debuggers%20And%20Tools-x86_en-us.msi'
ERROR: The remote file either doesn't exist, is unauthorized, or is forbidden for url 'https://codemachine.com/downloads/win10rs1/X86%20Debuggers%20And%20Tools-x86_en-us.msi'. Exception calling "GetResponse" with "0" argument(s): "The remote server returned an error: (404) Not Found."
The upgrade of windbg.flare was NOT successful.
Error while running 'C:\ProgramData\chocolatey\lib\windbg.flare\tools\chocolateyInstall.ps1'.
 See log for details.

Chocolatey upgraded 0/1 packages. 1 packages failed.
 See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log).

Failures
 - windbg.flare (exited 404) - Error while running 'C:\ProgramData\chocolatey\lib\windbg.flare\tools\chocolateyInstall.ps1'.
 See log for details.

**Comment by limbernie** [comment]
**windbg.flare** failed because the standalone installer was removed from `codemachine.com`. However, Wayback Machine (`web.archive.org`) still has a copy for both 32-bit and 64-bit as recent as 2019.

1. Get the download links from Wayback Machine. 
2. Make sure you cancel the download and get the actual download link.
3. Update the links in `\programdata\chocolatey\lib-bad\windbg.flare\tools\chocolateyInstall.ps1`
4. `cpack \programdata\chocolatey\lib-bad\windbg.flare`
5. `cinst --ignore-checksum windbg.flare -s \programdata\chocolatey\lib-bad\windbg.flare`

**Comment by darioongit** [comment]
do this change will included in the next release or these steps have to be executed at the end of the automation?

**Comment by lantuin** [comment]
>     5\. cinst --ignore-checksum

Is this solution working? I've updated the link (windbg.flare and also dnspy.flare) in the respective chocolateyInstall.ps1 but always the url is overwritten with the old one

**Comment by lantuin** [comment]
> 
> 
> **windbg.flare** failed because the standalone installer was removed from `codemachine.com`. However, Wayback Machine (`web.archive.org`) still has a copy for both 32-bit and 64-bit as recent as 2019.
> 
>     1. Get the download links from Wayback Machine.
> 
>     2. Make sure you cancel the download and get the actual download link.
> 
>     3. Update the links in `\programdata\chocolatey\lib-bad\windbg.flare\tools\chocolateyInstall.ps1`
> 
>     4. `cpack \programdata\chocolatey\lib-bad\windbg.flare`
> 
>     5. `cinst --ignore-checksum windbg.flare -s \programdata\chocolatey\lib-bad\windbg.flare`

Is this solution working? I've updated the link (windbg.flare and also dnspy.flare) in the respective chocolateyInstall.ps1 but always the url is overwritten with the old one

**Comment by lantuin** [comment]
> 
> 
> **windbg.flare** failed because the standalone installer was removed from `codemachine.com`. However, Wayback Machine (`web.archive.org`) still has a copy for both 32-bit and 64-bit as recent as 2019.
> 
>     1. Get the download links from Wayback Machine.
> 
>     2. Make sure you cancel the download and get the actual download link.
> 
>     3. Update the links in `\programdata\chocolatey\lib-bad\windbg.flare\tools\chocolateyInstall.ps1`
> 
>     4. `cpack \programdata\chocolatey\lib-bad\windbg.flare`
> 
>     5. `cinst --ignore-checksum windbg.flare -s \programdata\chocolatey\lib-bad\windbg.flare`

You missed .nuspec at the end of the filename!

cpack \programdata\chocolatey\lib-bad\windbg.flare.nuspec
cinst --ignore-checksum \programdata\chocolatey\lib-bad\windbg.flare.nuspec

**Comment by djjoa** [comment]
I added the x86 dependency that is missing to a GitHub project. Anyone can use this reference link from the comments above. 
Link : https://github.com/djjoa/commando-vm-dependency-fixes/raw/1d81e352379707aecbe8d93af3b1ae4964c2ddcc/X86%20Debuggers%20And%20Tools-x86_en-us.msi

### Solution to the OP's issue
Open powershell as admin
1. PS> `notepad.exe C:\programdata\chocolatey\lib-bad\windbg.flare\tools\chocolateyInstall.ps1`

2. change : 
        `$url = "https://codemachine.com/downloads/win10rs1/X86%20Debuggers%20And%20Tools-x86_en-us.msi"` 
TO 
        `$url = "https://github.com/djjoa/commando-vm-dependency-fixes/raw/1d81e352379707aecbe8d93af3b1ae4964c2ddcc/X86%20Debuggers%20And%20Tools-x86_en-us.msi" and save the file. 

AND 
        `$url64 = ""https://codemachine.com/downloads/win10rs1/X64%20Debuggers%20And%20Tools-x64_en-us.msi"`
TO 
        `$url64 = https://github.com/djjoa/commando-vm-dependency-fixes/raw/main/X64%20Debuggers%20And%20Tools-x64_en-us.msi"`

3. ` cpack C:\ProgramData\chocolatey\lib-bad\windbg.flare\windbg.flare.nuspec`
```powershell
----- output ---- 
PS C:\programdata\chocolatey\lib-bad\windbg.flare\tools> cpack C:\ProgramData\chocolatey\lib-bad\windbg.flare\windbg.flare.nuspec
Chocolatey v0.10.15
Attempting to build package from 'windbg.flare.nuspec'.
Successfully created package 'C:\programdata\chocolatey\lib-bad\windbg.flare\tools\windbg.flare.10.0.10586.22.nupkg'
PS C:\programdata\chocolatey\lib-bad\windbg.flare\tools>
```

4. `cinst --ignore-checksum \programdata\chocolatey\lib-bad\windbg.flare\windbg.flare.nuspec`
```powershell
----output ----- 
PS C:\programdata\chocolatey\lib-bad\windbg.flare\tools> cinst --ignore-checksum \programdata\chocolatey\lib-bad\windbg.flare\windbg.flare.nuspec
The use of .nupkg or .nuspec in for package name or source is known to cause issues. Please use the package id from the nuspec `<id />` with `-s .` (for local folder where nupkg is found).
Chocolatey v0.10.15
Installing the following packages:
\programdata\chocolatey\lib-bad\windbg.flare\windbg.flare.nuspec
By installing you accept licenses for the packages.
Attempting to build package from 'windbg.flare.nuspec'.
Successfully created package 'C:\programdata\chocolatey\lib-bad\windbg.flare\tools\windbg.flare.10.0.10586.22.nupkg'

windbg.flare v10.0.10586.22
windbg.flare package files install completed. Performing other installation steps.
File appears to be downloaded already. Verifying with package checksum to determine if it needs to be redownloaded.
WARNING: Ignoring checksums due to feature checksumFiles turned off or option --ignore-checksums set.
WARNING: Ignoring checksums due to feature checksumFiles turned off or option --ignore-checksums set.
C:\Users\commando\AppData\Local\Temp\windbg.installerx86.msi
Installing windbg.flare...
windbg.flare has been installed.
Downloading windbg.flare 64 bit
  from 'https://github.com/djjoa/commando-vm-dependency-fixes/raw/main/X64%20Debuggers%20And%20Tools-x64_en-us.msi'
Progress: 100% - Completed download of C:\Users\commando\AppData\Local\Temp\windbg.installerx64.msi (25.11 MB).
Download of windbg.installerx64.msi (25.11 MB) completed.
WARNING: Ignoring checksums due to feature checksumFiles turned off or option --ignore-checksums set.
C:\Users\commando\AppData\Local\Temp\windbg.installerx64.msi
Installing windbg.flare...
windbg.flare has been installed.
2021/02/06 20:10:13 [windbg.flare] chocolateyInstall.ps1 [+] INFO : Downloading symbols for: C:\Windows\System32\ntdll.dll
2021/02/06 20:10:15 [windbg.flare] chocolateyInstall.ps1 [+] INFO : Downloading symbols for: C:\Windows\System32\kernel32.dll
2021/02/06 20:10:16 [windbg.flare] chocolateyInstall.ps1 [+] INFO : Downloading symbols for: C:\Windows\System32\kernelbase.dll
2021/02/06 20:10:18 [windbg.flare] chocolateyInstall.ps1 [+] INFO : Downloading symbols for: C:\Windows\System32\user32.dll
2021/02/06 20:10:19 [windbg.flare] chocolateyInstall.ps1 [+] INFO : Downloading symbols for: C:\Windows\System32\advapi32.dll
2021/02/06 20:10:20 [windbg.flare] chocolateyInstall.ps1 [+] INFO : Downloading symbols for: C:\Windows\System32\gdi32.dll
2021/02/06 20:10:20 [windbg.flare] chocolateyInstall.ps1 [+] INFO : Downloading symbols for: C:\Windows\System32\comctl32.dll
2021/02/06 20:10:21 [windbg.flare] chocolateyInstall.ps1 [+] INFO : Downloading symbols for: C:\Windows\System32\comdlg32.dll
2021/02/06 20:10:22 [windbg.flare] chocolateyInstall.ps1 [+] INFO : Downloading symbols for: C:\Windows\System32\ws2_32.dll
2021/02/06 20:10:22 [windbg.flare] chocolateyInstall.ps1 [+] INFO : Downloading symbols for: C:\Windows\System32\netapi32.dll
2021/02/06 20:10:23 [windbg.flare] chocolateyInstall.ps1 [+] INFO : Downloading symbols for: C:\Windows\System32\ole32.dll
2021/02/06 20:10:25 [windbg.flare] chocolateyInstall.ps1 [+] INFO : Downloading symbols for: C:\Windows\System32\wininet.dll
2021/02/06 20:10:26 [windbg.flare] chocolateyInstall.ps1 [+] INFO : Downloading symbols for: C:\Windows\System32\msvcrt.dll
2021/02/06 20:10:27 [windbg.flare] chocolateyInstall.ps1 [+] INFO : Downloading symbols for: C:\Windows\System32\imm32.dll
2021/02/06 20:10:27 [windbg.flare] chocolateyInstall.ps1 [+] INFO : Downloading symbols for: C:\Windows\SysWow64\ntdll.dll
2021/02/06 20:10:29 [windbg.flare] chocolateyInstall.ps1 [+] INFO : Downloading symbols for: C:\Windows\SysWow64\kernel32.dll
2021/02/06 20:10:29 [windbg.flare] chocolateyInstall.ps1 [+] INFO : Downloading symbols for: C:\Windows\SysWow64\kernelbase.dll
2021/02/06 20:10:31 [windbg.flare] chocolateyInstall.ps1 [+] INFO : Downloading symbols for: C:\Windows\SysWow64\user32.dll
2021/02/06 20:10:31 [windbg.flare] chocolateyInstall.ps1 [+] INFO : Downloading symbols for: C:\Windows\SysWow64\advapi32.dll
2021/02/06 20:10:32 [windbg.flare] chocolateyInstall.ps1 [+] INFO : Downloading symbols for: C:\Windows\SysWow64\gdi32.dll
2021/02/06 20:10:33 [windbg.flare] chocolateyInstall.ps1 [+] INFO : Downloading symbols for: C:\Windows\SysWow64\comctl32.dll
2021/02/06 20:10:33 [windbg.flare] chocolateyInstall.ps1 [+] INFO : Downloading symbols for: C:\Windows\SysWow64\comdlg32.dll
2021/02/06 20:10:34 [windbg.flare] chocolateyInstall.ps1 [+] INFO : Downloading symbols for: C:\Windows\SysWow64\ws2_32.dll
2021/02/06 20:10:35 [windbg.flare] chocolateyInstall.ps1 [+] INFO : Downloading symbols for: C:\Windows\SysWow64\netapi32.dll
2021/02/06 20:10:36 [windbg.flare] chocolateyInstall.ps1 [+] INFO : Downloading symbols for: C:\Windows\SysWow64\ole32.dll
2021/02/06 20:10:38 [windbg.flare] chocolateyInstall.ps1 [+] INFO : Downloading symbols for: C:\Windows\SysWow64\wininet.dll
2021/02/06 20:10:39 [windbg.flare] chocolateyInstall.ps1 [+] INFO : Downloading symbols for: C:\Windows\SysWow64\msvcrt.dll
2021/02/06 20:10:40 [windbg.flare] chocolateyInstall.ps1 [+] INFO : Downloading symbols for: C:\Windows\SysWow64\imm32.dll
  windbg.flare may be able to be automatically uninstalled.
This is try 1/3. Retrying after 300 milliseconds.
 Error converted to warning:
 (32) The process cannot access the file because it is being used by another process: [\\?\C:\ProgramData\chocolatey\lib-bad\windbg.flare\tools]
This is try 2/3. Retrying after 400 milliseconds.
 Error converted to warning:
 (32) The process cannot access the file because it is being used by another process: [\\?\C:\ProgramData\chocolatey\lib-bad\windbg.flare\tools]
Maximum tries of 3 reached. Throwing error.
Attempted to delete bad package install path if existing. Had an error:
 (32) The process cannot access the file because it is being used by another process: [\\?\C:\ProgramData\chocolatey\lib-bad\windbg.flare\tools]
 The install of windbg.flare was successful.
  Software installed as 'msi', install location is likely default.

Chocolatey installed 1/1 packages.
```

Hope this helps!

**Comment by htnhan** [comment]
Folks, We have updated `windbg.flare`. It should now install properly. 

Thank you
